sim/warlock/pets.go

```diff
@@ -339,9 +351,14 @@ func (pet *WarlockPet) registerFireboltSpell() {
 
 		ManaCost: core.ManaCostOptions{BaseCost: 0.02},
 		Cast: core.CastConfig{
+			CD: core.Cooldown{
+				// emulate the imp 1.5s GCD that does not change with haste
+				Timer:    pet.NewTimer(),
+				Duration: 1500 * time.Millisecond,
+			},
 			DefaultCast: core.Cast{
 				GCD:      core.GCDDefault,
-				CastTime: time.Millisecond * 2500,
+				CastTime: 2500 * time.Millisecond,
```

Comment:
Instead of having this artificial cooldown you can make the cast IgnoreHaste: true and add a ModifyCast handler that manually applies the cast time to the cast. You can see how DKs Gargoyle has its Spell configured its exactly the same concept (spell is hasted but GCD cant get below 1.5)

Reply:
I guess I can change it for consistency but seems to achieve pretty much the same.

Reply:
Actually no; this is broken :joy:
The CD obviously only applies after the cast is finished so it's waiting 1.5s between every cast